packages/core/src/Dialog/Dialog.styled.tsx

```diff
@@ -0,0 +1,183 @@
+import type { IDialogProps } from '..'
+import { Box, CloseButton, Grid, Text, ThemeProvider } from '..'
+
+import * as Dialog from '@radix-ui/react-dialog'
+import { VisuallyHidden } from '@radix-ui/react-visually-hidden'
+import themeGet from '@styled-system/theme-get'
+import { motion } from 'framer-motion'
+import React from 'react'
+import styled from 'styled-components'
+
+export const DialogSizes = ['sm', 'md', 'lg', 'xl', 'full'] as const
+export type DialogSize = (typeof DialogSizes)[number]
+
+type SizeStyle = {
+  width?: string
+  height?: string
+  borderRadius?: string
+}
+
+const sizeStyles: Record<DialogSize, SizeStyle> = {
+  sm: {
+    width: 'calc(min(400px, 100vw) - 2 * 16px)',
+    borderRadius: '24px',
+  },
+  md: {
+    width: 'calc(min(640px, 100vw) - 2 * 16px)',
+    borderRadius: '24px',
+  },
+  lg: {
+    width: 'calc(min(960px, 100vw) - 2 * 16px)',
+    borderRadius: '24px',
+  },
+  xl: {
+    width: 'calc(min(1280px, 100vw) - 2 * 16px)',
+    height: 'calc(100% - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  full: {
+    width: '100%',
+    height: '100%',
+  },
+} as const
+
+const FloatingCloseButton: typeof CloseButton = styled(CloseButton)`
+  position: absolute;
+  top: ${(props) => -((props.dialogSize === 'full' || props.fullWidth ? 0 : 16) + 10)}px;
+  right: ${(props) => -((props.dialogSize === 'full' || props.fullWidth ? 0 : 16) + 10)}px;
+  margin: ${themeGet('space.3')};
+  padding: ${themeGet('space.2')};
+  background-color: ${themeGet('palette.background.lightest')};
+  box-shadow: ${themeGet('shadows.sm')};
+  &:hover {
+    background-color: ${themeGet('palette.background.lightest')};
+  }
+`
+
+const DialogOverlayWrapper = styled(motion.div)`
+  position: fixed;
+  inset: 0;
+  display: grid;
+  overflow-y: auto;
+  cursor: ${(props: IDialogProps) => (props.scrimDismiss ? 'pointer' : 'default')};
```

Comment:
I feel like the cursor should always be the default pointer. Using `pointer` might be confusing since it's unclear what the pointer is actually pointing at.

Reply:
The intention was to show that the scrim/overlay is clickable in order to be dismissed, but now looking at:
- [chakra](https://chakra-ui.com/docs/components/modal/usage)
- [bootstrap](https://getbootstrap.com/docs/4.0/components/modal/)
- [material-ui](https://mui.com/material-ui/react-modal/)
and even the radix dialog itself, you're right and this isn't a feature probably because the confusion you brought up! I will remove this 😄